packages/hebao_v1/contracts/base/BaseWallet.sol

```diff
@@ -232,30 +223,55 @@ contract BaseWallet is ReentrancyGuard, AddressSet, Wallet
         emit ModuleAdded(_module);
     }
 
+    receive() external payable { }
+
     /// @dev This default function can receive Ether or perform queries to modules
     ///      using bound methods.
     fallback()
         external
         payable
     {
-        if (msg.data.length == 0 || msg.value > 0) {
-            // Note: don't do anything here so send/transfer still works
-            // -- the 2300 gas limit for send/transfer may case out-of-gas
-            // issue after Istanbul.
-            return;
-        }
-
         address module = methodToModule[msg.sig];
         require(isAddressInSet(MODULE, module), "MODULE_UNAUTHORIZED");
 
-        (bool success, bytes memory returnData) = module.call(msg.data);
+        (bool success, bytes memory returnData) = nonReentrantCall(uint8(1), module, msg.value, msg.data);
```

Comment:
I don't think it's a good idea to make this `nonReentrant`. One of the reasons this functionality is here is so that a contract can call a function on `msg.sender` (which will be this wallet) which will also be the walllet so it is kind of made for reentrant by design. So if the transaction was initiated with `transact` this will fail. This can be a problem with ERC1271 or even token standards that call function on the receiver of tokens.
This function also does 0 state changes because it just forwards calls so I also don't think it's necessary. The functions called here are also only verified module functions which won't do much to begin with.

Reply:
OK